### ISSUE TITLE
AssertionError for Double Brace Initialization

### ISSUE THREAD

**Issue by islandsvinur** [cheptsov/AdvancedExpressionFolding#69]
I have a Java source file containing the following code:

```java
private static Set<String> fragmentBundles = new HashSet<String>() {{
    add("slf4j-simple-1.7.21.jar");
}};
```

When trying to open this with the Advanced Expression Folding plugin enabled, IntelliJ just silently ignores the file open request, showing the (abbreviated) stacktrace in the log included below. With the Advanced Expression Folding plugin disabled, the file opens without problems.

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.AssertionError: (1052,1052), text: new HashSet<String>() {{
        add("slf4j-simple-1.7.21.jar");
    }}, language = Language: JAVA
	at com.intellij.openapi.fileEditor.impl.text.AsyncEditorLoader.resultInTimeOrNull(AsyncEditorLoader.java:135)
  [...]
Caused by: java.util.concurrent.ExecutionException: java.lang.AssertionError: (1052,1052), text: new HashSet<String>() {{
        add("slf4j-simple-1.7.21.jar");
    }}, language = Language: JAVA
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
  [...]
Caused by: java.lang.AssertionError: (1052,1052), text: new HashSet<String>() {{
        add("slf4j-simple-1.7.21.jar");
    }}, language = Language: JAVA
	at com.intellij.lang.folding.FoldingDescriptor.<init>(FoldingDescriptor.java:106)
	at com.intellij.lang.folding.FoldingDescriptor.<init>(FoldingDescriptor.java:87)
	at com.intellij.lang.folding.FoldingDescriptor.<init>(FoldingDescriptor.java:72)
	at com.intellij.advancedExpressionFolding.SetLiteral$6.<init>(SetLiteral.java:81)
	at com.intellij.advancedExpressionFolding.SetLiteral.buildFoldRegions(SetLiteral.java:81)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:1642)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:1647)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:1647)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:1647)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:83)
	at com.intellij.lang.folding.CompositeFoldingBuilder.buildFoldRegions(CompositeFoldingBuilder.java:63)
  [...]
```

**Comment by islandsvinur** [comment]
Running version 0.9.0. Seems to be fixed in 0.9.2, for which a colleague got an update notification, but I didn't.

**Comment by islandsvinur** [comment]
My bad, added EAP repository and issue is indeed fixed in 0.9.2.
